### PR TITLE
chore(component-testing): Remove useless code & deps in @cypress/react and @cypress/vue

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
Somehow addresses #15108 bot for future.

Just cleaning up some unused code, our users have enough rights to decide do they need code coverage/modules mocking/fetch polyfilling or not. 

Also changing `renderTestingPlatform` logic to not duplicate the logic from `webpack-dev-server` that automatically create `__cypress_root` element.

### CODE REVIEW

File: npm/vue/src/support.js [hunk 1]
```diff
@@ -1,7 +1,4 @@
 /* eslint-env mocha */
```
Comment (elevatebart) [inline npm/vue/src/support.js:1]: I believe this whole file is useless.

**Comment by elevatebart** [comment]
Should we merge this into master directly to trigger new versions?

**Comment by elevatebart** [comment]
We have to fix all failing examples too...

**Review by lmiller1990 — DISMISSED** [review]
Great. But CI is 💀

File: npm/vue/src/support.js (hunk 1 above)
Comment (dmtrKovalenko) [reply npm/vue/src/support.js:1]: It's used for experimentalComponent testing

File: npm/react/package.json [hunk 2]
```diff
@@ -16,26 +16,21 @@
...
-    "@cypress/code-coverage": "3.8.6",
     "@cypress/webpack-preprocessor": "0.0.0-development",
-    "babel-plugin-istanbul": "6.0.0",
     "debug": "4.3.2",
-    "find-up": "5.0.0",
-    "find-webpack": "2.2.1",
-    "mime-types": "2.1.26",
-    "semver": "7.3.2",
-    "unfetch": "4.1.0"
+    "find-webpack": "2.2.1"
```
Comment (lmiller1990) [inline npm/react/package.json:21]: I think we should prob move away from `find-webpack` entirely eventually - it's only used in 4 places atm.

**Review by lmiller1990 — DISMISSED** [review]
nice reduction of tech debt. I hope we can remove find-webpack soon too, it's a little too magical for my liking

File: npm/react/package.json (hunk 2 above)
Comment (dmtrKovalenko) [reply npm/react/package.json:21]: @lmiller1990 we need some api for users to not look for webpack config manually for different options – create-react-app/next/nuxt. Once we will implement something fro this – we will be able to remove it :)

**Review by elevatebart — APPROVED** [review]
LGTM

**Pull request merged**